keps/sig-cli/3659-kubectl-apply-prune/README.md

```diff
@@ -627,128 +544,171 @@ In future, we may define additional “index” mechanisms here to further optim
-- `kubectl apply --prune --applyset=<id> --dry-run`
-- `kubectl diff --prune --applyset=<id>`
+- `kubectl apply --prune --applyset=[RESOURCE][.GROUP]/NAME`: The `--applyset` flag MUST be used with `--prune` and MUST have a non-empty value when used. Its GR component is defaulted to `secrets` when missing. This flag CANNOT be used with `-l/--selector` or with `--prune-allow-list`, and is validated at flag parsing time.
+- `kubectl apply --prune --applyset=[RESOURCE][.GROUP]/NAME --dry-run`
 
-Tentatively proposed for future iterations (more specific design details to follow after MVP):
-- `kubectl apply generate-applyset <id> --selector=[key=val] --legacy-allow-list=[]`: command to migrate from the legacy pruning system to this new one.
-- `kubectl apply verify-applyset [<id>] [--fix]`: `fsck`-style functionality to update the annotations on the parent ApplySet objects.
-- `kubectl apply view-applyset <id> -o name|json|yaml`: A command for viewing ApplySet membership, ideally in a way that can be programmatically chained.
-- `kubectl apply disband-applyset <id>`: removes the `applyset.k8s.io/id` from all members and then deletes the parent ApplySet object.
-- `kubectl apply list-applysets`: view apply sets, including those managed by other tools.
+To be included in v1.28:
+- `kubectl diff --prune --applyset=[RESOURCE][.GROUP]/NAME`
 
-Examples:
 
 ```bash
 # Simple namespace-scoped apply with prune.
 # The parent object will be a Secret named "set1" in the "foo" namespace.
 kubectl apply -n foo --prune --applyset=set1  -f .
 
-# Simple apply with prune, with cluster-scoped ApplySet
-# The parent object will be the "myapp" Namespace itself.
-kubectl apply -n myapp --prune --applyset=namespaces/myapp -f .
+# Simple apply with prune, with a ConfigMap parent object.
+# The parent object will be the "myapp" ConfigMap  in the "myapp" namespace.
+kubectl apply -n myapp --prune --applyset=configmaps/myapp -f .
 
-# Simple apply with prune, with cluster-scoped custom resource ApplySet
+# Simple apply with prune, with custom resource ApplySet
 # The parent object will be a VPA named "tracker" in the "foo" namespace.
 kubectl apply -n foo --prune --applyset=verticalpodautoscalers.autoscaling.k8s.io/tracker -f .
 
+# Simple apply with prune, with a cluster-scoped custom resource ApplySet
+# The parent object will be a MyCoApp named "my-app"
+kubectl apply -n foo --prune --applyset=mycoapps.platform.example.com/my-app -f .
+
 # Preview apply-with-prune
 kubectl apply -n foo --prune --applyset=set1 --dry-run -f .
 
 # Diff
 kubectl diff -n foo --prune --applyset=set1 -f .
+```
 
-# Optional commands follow:
-
-# Extension command to verify correspondence of annotations
-kubectl apply verify-applyset configmap/myset -n foo
+#### ApplySet subcommands
```

Comment:
Added more tentative details for discussion here, so we can start implementation of the ones we go with after code freeze. Please critique!